### PR TITLE
Properly handle files with spaces.

### PR DESCRIPTION
**Pull request by coryan**
Fixes the script in case we ever ran into files with spaces.

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/2179)

### CODE REVIEW

**Comment by coryan** [comment]
I think this addresses the issues raised in #2165.  I tested the script on my workstation by artificially inserting a directory with spaces in it, and it worked.  Naturally it also works for directories without spaces.

File: ci/travis/dump-reports.sh [hunk 1]
```diff
@@ -21,29 +21,40 @@ if [[ -z "${PROJECT_ROOT+x}" ]]; then
...
-  echo "No analysis reports found, exit scripts with success."
-  exit 0
-fi
+HAS_REPORTS=0
 
 # If w3m is installed there is nothing to do.
-if which w3m >/dev/null 2>&1; then
-  echo "Found w3m already installed."
+if type w3m >/dev/null 2>&1; then
+  /bin/true
```
Comment (devbww) [inline ci/travis/dump-reports.sh]: s:/bin/::, or just `:`, or `if ! type w3m ...`?

Comment (coryan) [reply ci/travis/dump-reports.sh]: Done.

File: ci/travis/dump-reports.sh [hunk 2]
```diff
@@ -21,29 +21,38 @@ if [[ -z "${PROJECT_ROOT+x}" ]]; then
...
+
+# Find any analysis reports, currently ABI checks and Clang static analysis are
+# the two things that produce them. Note that the Clang static analysis reports
+# are copied into the scan-build-output directory by the build-docker.sh script.
+
+find "${BUILD_OUTPUT}" -name 'compat_report.html' -print0 2>/dev/null | \
+    xargs -0 -I{} bash -c 'dump_report "{}"'
+
+find scan-build-output/ -name '*.html' -print0 2>/dev/null | \
+    xargs -0 -I{} bash -c 'dump_report "{}"'
```
Comment (remyabel) [inline ci/travis/dump-reports.sh]: Is there any reason not to use exec here? Something like

```shell
find scan-build-output/ -name '*.html' -exec dump_report "{}" \;
```

Comment (coryan) [reply ci/travis/dump-reports.sh]: Good idea. A straight `-exec` would not work, functions cannot be called that way.

**Comment by coryan** [comment]
PTAL.

File: ci/travis/dump-reports.sh [hunk 3]
```diff
@@ -46,13 +43,10 @@ export -f dump_report
...
-    xargs -0 -I{} bash -c 'dump_report "{}"'
+find "${BUILD_OUTPUT}" -name 'compat_report.html' \
+    -exec bash -c 'dump_report "{}"' \; 2>/dev/null || \
+  echo "No ABI compability reports found."
 
-if [[ "${HAS_REPORTS}" = 0 ]]; then
-  echo "No analysis reports found, exit scripts with success."
-  exit 0
-fi
+find scan-build-output/ -name 'compat_report.html' \
```
Comment (devbww) [inline ci/travis/dump-reports.sh]: s/compat_report/*/?

File: ci/travis/dump-reports.sh [hunk 4]
```diff
@@ -46,13 +43,10 @@ export -f dump_report
...
+    -exec bash -c 'dump_report "{}"' \; 2>/dev/null || \
```
Comment (devbww) [inline ci/travis/dump-reports.sh]: You should probably reverse the quoting here:

`-exec bash -c "dump_report '{}'" \;`

Consider, for example, the presence of a 'scan-build-output/${USER}.html' file.

File: ci/travis/dump-reports.sh (hunk 3 above)
Comment (remyabel) [inline ci/travis/dump-reports.sh]: I think you accidentally changed `*.html` to `compat_report`.

Comment (coryan) [reply ci/travis/dump-reports.sh]: Doh! Thanks!

Comment (coryan) [reply ci/travis/dump-reports.sh]: Yup. Fixed.

File: ci/travis/dump-reports.sh (hunk 4 above)
Comment (coryan) [reply ci/travis/dump-reports.sh]: Yup.

**Pull request merged**